backend/macros/src/config.rs

```diff
@@ -189,10 +189,23 @@ fn gen_root_mod(input: &Input, visibility: &TokenStream) -> TokenStream {
 
             let try_from_fields = collect_tokens(children, |node| {
                 match node {
-                    Node::Leaf { name, .. } => {
-                        let path = format!("{}.{}", path.join("."), name);
-                        quote! {
-                            #name: src.#name.ok_or(self::util::TryFromError { path: #path })?,
+                    Node::Leaf { name, ty, .. } => {
+                        match as_option(ty) {
+                            // If this value is optional, we just move it as it can never fail.
+                            Some(_) => quote! { #name: src.#name, },
+
+                            // Otherwise, we return an error if the value hasn't been specified.
+                            None => {
+                                let path = match path.is_empty() {
+                                    true => name.to_string(),
+                                    false => format!("{}.{}", path.join("."), name),
+                                };
```

Comment:
Is `match`-ing on a `bool` (especially over a simple `if`) really idiomatic? :eyes:

Reply:
It's not non-idiomatic. It depends. If the "body" is only a short expression like here, then I use `match`. Otherwise, `if-else` is preferred, yes.

Reply:
Ok, sure, why not.<sub>(BECAUSE IT LOOKS HECKING WEIRD! :scream:)</sub>